get_info.php

```diff
@@ -1 +1,295 @@
+    exit();
+}
+
+function get_query_result($query_string, $bind_variables)
+{
+    require '../db/config.php';
+    // Attempt to execute sql command and print response in json format.
+    // If a sql error occurs, JSON error object.
+    try {
+        // connect to the mysql database
+        $conn = new PDO(
+            "mysql:host=$servername;dbname=$database",
+            $username,
+            $password
+        );
+        // set the PDO error mode to exception
+        $conn->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+        $stmt = $conn->prepare($query_string);
+        foreach ($bind_variables as $key => $value) {
+            $stmt->bindValue($key, $value);
+        }
+        $stmt->execute();
+
+        //fetch all results
+        $fetched = $stmt->fetchAll(PDO::FETCH_ASSOC);
+        return $fetched;
+    } catch (PDOException $e) {
+        $result = set_http_response(500);
+        date_default_timezone_set('America/Toronto');
+        error_log(
+            date("Y-m-d h:i:sa") . " : " . $e->getMessage() . "\n",
+            3,
+            "errors.log"
+        );
+        print json_encode($result, JSON_PRETTY_PRINT);
+        exit();
+    }
+}
+
+/**
+ * This function handles GET requests and returns a sql query that gets the
+ * required data for the what according to the API documentation.
+ *
+ * @throws InvalidPage The requested what does not exist
+ * @return sql query to the corresponding what parameter
+ */
+function handle_get($parameters)
+{
+    $bind_variables = [];
+    $role = [];
+    if (isset($parameters['user_id'])) {
+        $bind_variables[':user_id'] = $parameters['user_id'];
+        $role = get_query_result(gen_query_user_role(), $bind_variables);
+    }
+    if (isset($parameters['survey_id'])) {
+        $bind_variables[':survey_id'] = $parameters['survey_id'];
+    }
+    if (isset($parameters["term"])) {
+        $bind_variables[':term'] = $parameters["term"];
+    }
+    if (isset($parameters["course_code"])) {
+        $bind_variables[':course_code'] = $parameters["course_code"];
+    }
+    switch ($parameters["what"]) {
+        case "tas":
+            $tas = get_query_result(
+                set_parameters($parameters),
+                $bind_variables);
+
+            $ta_package = array('TYPE' => "ta_package", 'DATA' => $tas);
+            return $ta_package;
+        case "questions":
+            $questions = get_query_result(
+                gen_query_questions(),
+                $bind_variables
+            );
+            $question_package = array(
+                'TYPE' => "questions_package",
+                'DATA' => $questions
+            );
+            return $question_package;
+        case "course_pairings":
+            if($parameters["column"]=="sections"){
+                $course_pairings = get_query_result(
+                    gen_query_course_pairings_section(),
+                    $bind_variables
+                );
+            }else{
```

Comment:
Let's format our sourcecode....

Reply:
Sorry. Still a work in progress. I'll squash these 2 commits afterwords. Please don't merge.